mars/tensor/expressions/arithmetic/core.py

```diff
@@ -220,6 +176,27 @@ def casting(self):
     def err(self):
         return getattr(self, '_err', dict())
 
+    def _set_inputs(self, inputs):
+        super(TensorBinOp, self)._set_inputs(inputs)
+        inputs_iter = iter(self._inputs)
+
+        if getattr(self, '_lhs', None) is None:
+            # create binop from beginning
+            self._lhs = next(inputs_iter)
+            self._rhs = next(inputs_iter)
+            if inputs[2] is not None:
```

Comment:
Similar question to clip.py. iter() and getitem should not be mixed in case of ambiguity. This may apply to all ``_set_inputs()`` changed in this PR.